src/Cake.Core/Scripting/ScriptReference.cs

```diff
@@ -0,0 +1,59 @@
+﻿using System;
+using System.Globalization;
+using System.Security.Cryptography;
+using System.Text;
+using Cake.Core.IO;
+
+namespace Cake.Core.Scripting
+{
+    /// <summary>
+    /// Unique Script Reference
+    /// </summary>
+    public class ScriptReference
+    {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="ScriptReference" /> class.
+        /// </summary>
+        /// <param name="localPath">The local path of the </param>
+        /// <param name="origin">The original file path or URI of the script.</param>
+        public ScriptReference(FilePath localPath, string origin = null)
+        {
+            if (localPath == null)
+            {
+                throw new ArgumentNullException("localPath");
+            }
+
+            Path = localPath;
+
+            if (origin == null)
+            {
+                origin = localPath.ToString();
+            }
+            Origin = origin;
+
+            var hasher = new SHA1CryptoServiceProvider();
```

Comment:
is calculating a hash for the path really necessary? Wouldn't the path be enough as an ID?